userapi/src/main/java/model/IDataModel.java

```diff
@@ -0,0 +1,57 @@
+package model;
+
+import fk.prof.storage.AsyncStorage;
+
+import java.util.Set;
+
+/**
+ * Interface for DataStores containing aggregated profile data
+ * Created by rohit.patiyal on 23/01/17.
+ */
+public interface IDataModel {
+    /**
+     * Initializes the DataModel with DataStorage {@link AsyncStorage}
+     *
+     * @param store datastore object
+     */
+    void setStorage(AsyncStorage store);
```

Comment:
No, the interface only tells you the about the capability. There should not be a method related to "how?".
interface is usually related to 'what' and the impl to 'how'.